### ISSUE TITLE
javascript code is messy in Subsonic

### ISSUE THREAD

**Issue by biconou** [Libresonic/libresonic#332]
I am working on some piece of javascript code (in playQueue.jsp) and I think it is very messy.
I think that we should discuss of some guidelines to write a better javascript code in our Libresonic evolutions.

- javascript is mixed with jsp tags.
example in playQueue.jsp
```
    function onSkip(index) {
    <c:choose>
    <c:when test="${model.player.web}">
        skip(index);
    </c:when>
    <c:otherwise>
        currentStreamUrl = songs[index].streamUrl;
        playQueueService.skip(index, playQueueCallback);
    </c:otherwise>
    </c:choose>
    }
```
 
- javascript should be stored in separate .js files and shrinked
- jQuery core is misused
Example 
```
        if ($("#start")) {
            $("#start").toggle(!playQueue.stopEnabled);
            $("#stop").toggle(playQueue.stopEnabled);
        }
```
This will not test if an element whose id is "start" exists. One should write $("#start").length > 0
see https://learn.jquery.com/using-jquery-core/faq/how-do-i-test-whether-an-element-exists/

**Comment by fxthomas** [comment]
Seconded. IMO a lot of this will need to be redesigned anyway when doing #45.

For instance, the way we check for different player types (jukebox, cast, web) felt very hacky last time I touched this.

**Comment by Tapchicoma** [comment]
Just my side note: [Kotlin lang](https://github.com/JetBrains/kotlin) supports compiling into JS and can interop with existing JS libraries. It may be useful to write all the stuff in it and may be easier for java developers (kotlin can be learned by java dev pretty fast) to write js code as well.

**Comment by biconou** [comment]
I  would be very enthusiast at using Kotlin.

**Comment by biconou** [comment]
@Tapchicoma I am also interested in Vaadin to avoid all these messy JS things.

**Comment by muff1nman** [comment]
I'm against deviating too far from the JavaScript language. In my experience it complicates things more than it helps.

**Comment by biconou** [comment]
@muff1nman Indeed, many people say that. But there are still many initiatives to transpil javascript.
